### ISSUE TITLE
Bug in RawDataParser.php when a row has no data

### ISSUE THREAD

**Issue by KeanuTang** [smalot/pdfparser#679]
- PHP Version: 8.1
 - PDFParser Version: 2.8.0

### PDF input
The PDF has personal identifying information that I cannot share through a post like this. I could maybe email it directly if necessary.

### Expected output & actual output
I'm just trying to call 

### Code
```
                            case 'pdf':
                                try {
                                    $parser = new Parser([], $config);
                                    $PDF = $parser->parseFile($filepath);
                                    $pdfText = mb_convert_encoding($PDF->getText(), 'UTF-8');
                                    $result .= $pdfText;
                                } catch (\Exception $e) {
                                    \Log::error('Error parsing recommendation PDF text: '.$e->getMessage());
                                    break;
                                }
```

I'm getting this error:

![image](https://github.com/smalot/pdfparser/assets/121991155/1b2746b9-395c-493d-9014-1fa0b1cee89b)

In \smalot\pdfpaser\src\Smalot\PdfParser\RawData\RawDataParser.php, the issue seems to be coming from this code:
```
            } else {
                // number of bytes in a row
                $rowlen = array_sum($wb);
                // convert the stream into an array of integers
                $sdata = unpack('C*', $xrefcrs[1][3][0]);
                // split the rows
                $ddata = array_chunk($sdata, $rowlen);
            }
```

I wonder if it should be something like this?
```
            } else {
                // number of bytes in a row
                $rowlen = array_sum($wb);
                if ($rowlen > 0) {
                    // convert the stream into an array of integers
                    $sdata = unpack('C*', $xrefcrs[1][3][0]);
                    // split the rows
                    $ddata = array_chunk($sdata, $rowlen);
                } else {
                    $dddata = [];
                }
            }
```

**Comment by KeanuTang** [comment]
I'm very new to this code and library, but the code snippet comes from:
\smalot\pdfpaser\src\Smalot\PdfParser\RawData\RawDataParser.php

**Comment by GreyWyvern** [comment]
I haven't really worked with **RawDataParser.php** much, and a lot of what it does is still voodoo to me. The issue is with the ```/W [0 0 0]``` commands near the bottom of the file @KeanuTang just posted. Even after reading the PDF Reference (v1.7 pp 108-109) concerning this command, I'm still not completely wrapping my head around it.

In any case, the PDF Reference says the following:

> A value of zero for an element in the **W** array indicates that the corresponding field is not present in the stream, and the default value is used, if there is one. If the first element is zero, the type field is not present, and it defaults to type 1.

So it seems to me that a ```/W [0 0 0]``` command is indicating that *none* of the fields are present, and that this stream is essentially "empty". There *are* eight bytes in the associated streams, but with "field lengths" of all zeros, there is no way to know how to chunk or decode it. They're probably just placeholder bytes IMHO.

I think we're fine to return an empty array in that case, as in @KeanuTang's suggested code.

**Comment by KeanuTang** [comment]
Sounds good to me. I haven't spent much time looking at **RawDataParser.php** myself. I just investigated where the error was coming from and saw the code above. I'll trust you all to do what's best for the **pdfparser** code.

I've removed the PDF file. Just let me know if you need it again.

**Comment by KeanuTang** [comment]
@GreyWyvern Did you get the file I uploaded from before? I think this was the problematic file. I didn't create the PDF, so I'm not sure how it was first generated.

[file.pdf](https://github.com/smalot/pdfparser/files/14450814/file.pdf)

**Comment by GreyWyvern** [comment]
> @GreyWyvern Did you get the file I uploaded from before? I think this was the problematic file. I didn't create the PDF, so I'm not sure how it was first generated.

Yes, I have it, but you've mentioned we can't use it because it contains some sensitive information, which is fine. I've also tried editing your file by removing all the text, but any client I use to save it deletes the `/w [0 0 0]` XRef's that are causing this. Acrobat actually hangs on me when trying to load your file, even though you can still read it, which is weird.

If possible, you might want to contact the individual who created this file and sent it to you, ask them to create a new blank document in the same way, save it and send it back to you.

It is just one single test for a zero condition, so it might be okay to just commit it without a unit test.
